src/mxcp/server/admin/endpoints/endpoints.py

```diff
@@ -70,26 +76,38 @@ async def list_endpoints() -> EndpointListResponse:
                 elif endpoint_def:
                     # Extract endpoint data from definition
                     endpoint_type: Literal["tool", "resource", "prompt"] | None = None
-                    endpoint_data = None
                     enabled = True
 
-                    if "tool" in endpoint_def:
+                    name: str | None
+                    description: str | None
+                    language: str | None
+
+                    if endpoint_def.tool is not None:
+                        tool_def: ToolDefinitionModel = endpoint_def.tool
                         endpoint_type = "tool"
-                        endpoint_data = endpoint_def["tool"]
-                    elif "resource" in endpoint_def:
+                        name = tool_def.name
+                        description = tool_def.description
+                        language = tool_def.language
+                        enabled = tool_def.enabled
+                    elif endpoint_def.resource is not None:
+                        resource_def: ResourceDefinitionModel = endpoint_def.resource
                         endpoint_type = "resource"
-                        endpoint_data = endpoint_def["resource"]
-                    elif "prompt" in endpoint_def:
+                        name = resource_def.name or resource_def.uri
+                        description = resource_def.description
+                        language = resource_def.language
+                        enabled = resource_def.enabled
+                    elif endpoint_def.prompt is not None:
+                        prompt_def: PromptDefinitionModel = endpoint_def.prompt
                         endpoint_type = "prompt"
-                        endpoint_data = endpoint_def["prompt"]
-
-                    if endpoint_data:
-                        enabled = bool(endpoint_data.get("enabled", True))
-
-                    # Extract metadata
-                    name = endpoint_data.get("name") if endpoint_data else None
-                    description = endpoint_data.get("description") if endpoint_data else None
-                    language = endpoint_data.get("language") if endpoint_data else None
+                        name = prompt_def.name
+                        description = prompt_def.description
+                        language = None
```

Comment:
`language = None` makes me think our `prompt` feature does actually have a language (hardcoded as jinja, and source code is entered in the .yml file). Technically jinja could be one of the languages prompts can be defined in. Perhaps there is a case for prompts implemented in python (a function returning string) if one would want to customize the content of a prompt with global info picked from Salesforce before it's published. Just a thought.